### PR TITLE
Update Gsoc Guide for 2022

### PR DESCRIPTION
**Pull request by BenLloydPearson**
#### Summary
Updating Google Summer of Code guide for 2022

#### Ticket Link
n/a

### CODE REVIEW

File: site/content/contribute/gsoc/_index.md [hunk 1]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
 description: "Ideas list for Google Summer of Code"
 ---
 
-# GSoC 2021 Ideas
+# Google Summer of Code
+This page outlines everything you need to know about contributing to Mattermost for the Google Summer of Code (GSoC).
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:7]: Do we need the article? Not sure, actually, so this is more a question than a request for changes :)
```suggestion
This page outlines everything you need to know about contributing to Mattermost for Google Summer of Code (GSoC).
```

File: site/content/contribute/gsoc/_index.md [hunk 2]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
+
+## Contributor Guidance
+1. Read everything in this doc, we've put this together to maximize your chances of drafting a successful GSoC proposal.
+1. Learn about our [contribution process](https://developers.mattermost.com/contribute/getting-started/). We encourage applicants to participate in one of them many contribution opportunities we have prior to applying.
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:11]: Typo?
```suggestion
1. Learn about our [contribution process](https://developers.mattermost.com/contribute/getting-started/). We encourage applicants to participate in one of the many contribution opportunities we have prior to applying.
```

File: site/content/contribute/gsoc/_index.md [hunk 3]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
+1. Review our idea list below for any that might that interest you, all of the details are listed in GitHub Issues on our Community repository.
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:12]: Typo? Also, maybe we can link at the whole list here
```suggestion
1. Review our idea list below for any that might interest you, all of the details are listed [in GitHub Issues on our Community repository](https://github.com/mattermost/community/issues?q=is%3Aissue+is%3Aopen+%22GSoC+Idea%3A%22).
```

File: site/content/contribute/gsoc/_index.md [hunk 4]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
+1. Communicate! 
+   * Sign up for an account on our [community collaboration platform](https://community.mattermost.com). Join the [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code) to chat with mentors and other contributors.
+   * Leave any questions you have about the projects as comments on the GitHub issues linked below.
+   * If you have general questions about participating in GSoC with Mattermost, send an email to gsoc@mattermost.com
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:16]: Should we direct them to the channel first?
```suggestion
   * If you have general questions about participating in GSoC with Mattermost, ask in the [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code) or send an email to gsoc@mattermost.com
```

File: site/content/contribute/gsoc/_index.md [hunk 5]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
+1. Work with your mentors on your application, all applications must go through Google's official application process.
+1. Sumit your application before the deadline.
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:18]: These read a bit weird. What about this?
```suggestion
1. Work with the project mentors on your application: they are there to help you write a good proposal!
1. Submit your application before the deadline. Note that all applications must go through Google's official application process.
```

File: site/content/contribute/gsoc/_index.md [hunk 6]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
+Here are our mentors for this year, the links lead to their profiles on our community server
 
--   Four Software Design Engineers who will provide technical mentorship for the projects:
+-   Technical Engineering Mentors:
     -   Devin Binnie ([@devin.binnie](https://community.mattermost.com/core/messages/@devin.binnie))
     -   Daniel Espino García ([@daniel.espino.garcia](https://community.mattermost.com/core/messages/@daniel.espino.garcia))
     -   Alejandro García Montoro ([@alejandro.garcia](https://community.mattermost.com/core/messages/@alejandro.garcia))
-    -   Martin Kraft ([@martin.kraft](https://community.mattermost.com/core/messages/@martin.kraft))
--   A Product Manager who will provide guidance, direction, and support across all of the projects:
+    -   Elias N ([@elias.nahum](https://community.mattermost.com/core/messages/@elias.nahum))
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:29]: Poor Elias does not have his last name!
```suggestion
    -   Elias Nahum ([@elias.nahum](https://community.mattermost.com/core/messages/@elias.nahum))
```

File: site/content/contribute/gsoc/_index.md [hunk 7]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
-
--   **Expected outcomes**: The design and implementation of at least two new locations. This project can be adapted to the student to implement as many locations as desired.
--   **Skills required**: React, React Native
--   **Difficulty**: Medium
--   **Mentors**: Daniel Espino García, Alejandro García Montoro
+All project ideas are listed as issues on our community GitHub repo. Here are the summaries:
+* [Playbooks mobile app](https://github.com/mattermost/community/issues/171) - Build a proof of concept Playbooks integration for the Mattermost mobile app.
+* [Boards mobile app](https://github.com/mattermost/community/issues/170) - Build a proof of concept Boards integration for the Mattermost mobile app.
+* [CI pipeline analysis](https://github.com/mattermost/community/issues/169) - Publish a plugin to analyze the CI pipelines in Mattermost to alert the team of build and test failures
+* [Speech to Text Integration](https://github.com/mattermost/community/issues/168) - Publish a plugin to analyze the CI pipelines in Mattermost to alert the team of build and test failures
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:43]: The descriptions of these are duplicated.

File: site/content/contribute/gsoc/_index.md [hunk 8]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
+* [Post Edit History](https://github.com/mattermost/community/issues/167) - Publish a plugin to view the history of edits in posts and improve message history auditing
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:44]:
```suggestion
* [Post Edit History](https://github.com/mattermost/community/issues/167) - Publish a plugin to view the history of edits in posts and improve message history auditing.
```

File: site/content/contribute/gsoc/_index.md [hunk 9]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
+* [Recordings for Calls](https://github.com/mattermost/community/issues/166) - Design and implement a recordings service for Calls.
+* [Qualify posts importance](https://github.com/mattermost/community/issues/165) - Design and implement a system that introduces the concept of importance to Mattermost messages.
+* [Post timed reminders](https://github.com/mattermost/community/issues/164) - Publish a plugin that manages timed reminders on posts that should be read later.
+* [Username aliases](https://github.com/mattermost/community/issues/163) - Design and implement a system to manage username aliases.
+* [Custom keyboard shortcuts](https://github.com/mattermost/community/issues/162) - Design and implement a system to let users customize their keyboard shortcuts
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:49]:
```suggestion
* [Custom keyboard shortcuts](https://github.com/mattermost/community/issues/162) - Design and implement a system to let users customize their keyboard shortcuts.
```

File: site/content/contribute/gsoc/_index.md [hunk 10]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
 
-### Bring your own idea
 
-If none of the ideas in this list speak to you, you can always propose your own idea!
+### Submit Your Own Idea
+If none of the ideas in this list speak to you, you can always propose new ones! Maybe you’ve been wanting to build a specific plugin for a while, you have an amazing core feature in mind that would bring the product to a whole new level, or you want to deep dive into the performance of the server to further optimize it. 
 
-Maybe you’ve been wanting to build a specific plugin for a while, you have an amazing core feature in mind that would bring the product to a whole new level, or you want to deep dive into the performance of the server to further optimize it.
+Ideas proposed by students are more than welcome, don’t hesitate to send them our way! As with all the other projects, you need to take into account that the proposals would need to fit in the overall philosophy of the project: Our product managers will take a look before we can approve them, and we can always discuss the details with the whole team if we decide to move forward with it.
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:57]: This year they might not be students :)
```suggestion
Ideas proposed by contributors are more than welcome, don’t hesitate to send them our way! As with all the other projects, you need to take into account that the proposals would need to fit in the overall philosophy of the project: Our product managers will take a look before we can approve them, and we can always discuss the details with the whole team if we decide to move forward with it.
```

File: site/content/contribute/gsoc/_index.md [hunk 11]
```diff
@@ -3,224 +3,58 @@ title: "Google Summer of Code"
...
 
-Ideas proposed by students are more than welcome, so don’t hesitate to send them our way! As with all the other projects, proposals need to fit in the overall philosophy of the project. Our Product Management team reviews each idea prior to approval, and we can always discuss the details with the whole team if we decide to move forward with it.
+If you want to propose your own idea, share your idea as early as possible in our [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code
+).
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:60]: Formatting:
```suggestion
If you want to propose your own idea, share your idea as early as possible in our [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code).
```

**Comment by agarciamontoro** [comment]
Whoops, and the title seems duplicated (the title of the page and the first heading are the same). What do you think if we have two h1 headings: the first would be "Contributor guidance" (or something similar) and the second one would be "Ideas list". Thoughts?
image

**Comment by BenLloydPearson** [comment]
@agarciamontoro I committed all of your suggested changes and fixed the duplicated title.

File: site/content/contribute/gsoc/_index.md [hunk 12]
```diff
@@ -2,225 +2,56 @@
...
 
--   Four Software Design Engineers who will provide technical mentorship for the projects:
+-   Technical Engineering Mentors:
     -   Devin Binnie ([@devin.binnie](https://community.mattermost.com/core/messages/@devin.binnie))
     -   Daniel Espino García ([@daniel.espino.garcia](https://community.mattermost.com/core/messages/@daniel.espino.garcia))
     -   Alejandro García Montoro ([@alejandro.garcia](https://community.mattermost.com/core/messages/@alejandro.garcia))
-    -   Martin Kraft ([@martin.kraft](https://community.mattermost.com/core/messages/@martin.kraft))
--   A Product Manager who will provide guidance, direction, and support across all of the projects:
+    -   Elias Nahum ([@elias.nahum](https://community.mattermost.com/core/messages/@elias.nahum))
+    -    Claudio Costa ([@claudio.costa](https://community.mattermost.com/core/messages/@claudio.costa))
```
Comment (agarciamontoro) [inline site/content/contribute/gsoc/_index.md:28]: Just one more addition and I think we can merge! :tada:
```suggestion
    -    Claudio Costa ([@claudio.costa](https://community.mattermost.com/core/messages/@claudio.costa))
    -    Ibrahim Acikgoz ([@ibrahim.acikgoz](https://community-daily.mattermost.com/core/messages/@ibrahim.acikgoz))
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Looking great, thank you for addressing the changes!

**Review by agarciamontoro — APPROVED** [review]
:tada:

**Review by streamer45 — APPROVED** [review]
Looks great :+1: 

My only suggestion would be to possibly ask for a quick pass from our writers team. Can be done post merge if this needs to go in asap.

**Comment by agarciamontoro** [comment]
@streamer45, yup, I'll ask @cwarnermm or @justinegeffen for a review post-merge! We want to have it merged now, so we can review it post-merge. 

Thanks for the review! :tada:

**Pull request merged**